File: 0671-second-minimum-node-in-a-binary-tree/0671-second-minimum-node-in-a-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void dfs(TreeNode* root, set<int>& set){
        if(!root) return;
        set.insert(root->val);
        
        dfs(root->left, set);
        dfs(root->right, set);
    }
    int findSecondMinimumValue(TreeNode* root) {
        set<int> set;
        dfs(root, set);
        
        int i = 0; 
        for(auto& it: set){
            i++;
            if(i == 2) return it;
        }

        return -1;
    }
};

File: 0671-second-minimum-node-in-a-binary-tree/0671-second-minimum-node-in-a-binary-tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def dfs (self, root, list_set):
        if not root:
            return
        list_set.add(root.val)
        self.dfs(root.left, list_set)
        self.dfs(root.right, list_set)
    
    def findSecondMinimumValue(self, root):
        """
        :type root: Optional[TreeNode]
        :rtype: int
        """
        list_set = set()
        self.dfs(root, list_set)

        #sorted(list_set)

        if len(list_set) < 2:
            return -1
        
        return sorted(list_set)[1]
        